consensus/polybft/contractsapi/bindings-gen/main.go

```diff
@@ -0,0 +1,293 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"text/template"
+
+	gensc "github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi/artifact"
+	"github.com/umbracle/ethgo/abi"
+)
+
+const (
+	contractVariableName   = "%sContract"
+	contractStructName     = "%sContractImpl"
+	contractVariableFormat = "%s = &%s{Artifact: %s}"
+	abiTypeNameFormat      = "var %sABIType = abi.MustNewType(\"%s\")"
+	eventNameFormat        = "%sEvent"
+)
+
+func main() {
+	cases := []struct {
```

Comment:
Can this be dynamic instead of hardcoded, based on the contract?

Reply:
It can, but, we don't need all the functions and events from contracts, plus we don't need to generate stubs from all contracts as well. I would say this is ok, because that artifact generator that Boris did also has a similar thing.

Reply:
I won't be selective on this, TBH I prefer that it generates everything, we use only what we need.

Reply:
I am fine with both. @ferranbt what do you think?

Reply:
I'm more with Victor's suggestion here and generate everything if that's not going to overcomplicate logic. 
For example, it is a question of why we don't have generated bindings for staking/unstaking and delegation/undelegation. We could use those bindings from the helper commands we have. 

Therefore it is probably better to generate everything and use whatever subset we need.

Reply:
It is described in the RFC section for this change. The bindings for staking/unstaking methods for example interact with the contracts over JsonRPC while the ones we are covering here interact with the contract as `Transaction` objects inside `Polybft`, that is why we only need a small subset of them.

The other Go bindings should be generated with [`abigen`](https://github.com/umbracle/ethgo/tree/main/cmd/abigen).

Reply:
@ferranbt This will require maintenance in the future, adding or removing new parts, if we generate everything regardless of what we use, it's just gen and forget. I would prefer to generate everything.